code/parachain/frame/pablo/src/dual_asset_constant_product.rs

```diff
@@ -139,7 +141,13 @@ impl<T: Config> DualAssetConstantProduct<T> {
 				keep_alive,
 			)?;
 
-			single_deposit.value
+			(
+				single_deposit.value,
+				assets_with_balances
```

Comment:
NIT: Additional iterations to be removed

Reply:
what do you mean by iterations? `assets_with_balances` is a `Vec<AssetAmount>` and we need a `BTreeMap<AssetId, Balance>`